tests/reprsimil/test_gbrsa.py

```diff
@@ -0,0 +1,364 @@
+        s2XTAcorrX[i] = np.dot(design_mat[i].T, design_mat[
+                               i]) * s[:, None, None]**2 * a[:, None, None]
+        sXTAcorrY[i] = np.dot(design_mat[i].T, Y[i]) * \
+            s[:, None, None] * a[:, None, None]
+
+    # test if the gradients are correct
+    print(log_fixed_terms)
+    ll0, deriv0 = gbrsa._sum_loglike_marginalized(L_vec, s2XTAcorrX, YTAcorrY_diag,
+                                                  sXTAcorrY, half_log_det_X0TAX0,
+                                                  log_weights, log_fixed_terms,
+                                                  l_idx, n_C, n_T, n_V, n_X0,
+                                                  n_grid, rank=None)
+    # We test the gradient to the Cholesky factor
+    vec = np.random.randn(np.size(L_vec))
+    vec = vec / np.linalg.norm(vec)
+    dd = nd.directionaldiff(lambda x: gbrsa._sum_loglike_marginalized(x, s2XTAcorrX, YTAcorrY_diag,
+                                                                      sXTAcorrY, half_log_det_X0TAX0,
+                                                                      log_weights, log_fixed_terms,
+                                                                      l_idx, n_C, n_T, n_V, n_X0,
+                                                                      n_grid, rank=None)[0],
+                            L_vec, vec)
+    assert np.isclose(dd, np.dot(deriv0, vec), rtol=1e-5), 'gradient incorrect'
+
+
+def test_SNR_grids():
+    import brainiak.reprsimil.brsa
+    import numpy as np
+
+    s = brainiak.reprsimil.brsa.GBRSA(SNR_prior='unif', SNR_bins=10)
+    SNR_grids, SNR_weights = s._set_SNR_grids()
+    assert np.isclose(np.sum(SNR_weights), 1) and np.isclose(np.std(SNR_weights[1:-1]), 0) and np.all(SNR_weights > 0)\
+        and np.isclose(np.min(SNR_grids), 0) and np.all(SNR_grids >= 0) and np.isclose(np.max(SNR_grids), 1), \
+        'SNR_weights or SNR_grids are incorrect for uniform prior'
+    assert np.isclose(np.std(np.diff(SNR_grids[1:-1])), 0), \
+        'SNR grids are not equally spaced for uniform prior'
+    assert np.size(SNR_grids) == np.size(SNR_weights) and np.size(SNR_grids) == 10 ,\
+        'size of SNR_grids or SNR_weights is not correct for uniform prior'
+
+
+    s = brainiak.reprsimil.brsa.GBRSA(SNR_prior='lognorm', SNR_bins=35)
+    SNR_grids, SNR_weights = s._set_SNR_grids()
+    assert np.all(SNR_grids >= 0) and np.isclose(np.sum(SNR_weights), 1) and np.all(SNR_weights > 0) \
+        and np.isclose(np.min(SNR_grids), 0) and np.all(np.diff(SNR_grids) > 0), \
+        'SNR_grids or SNR_weights not correct for log normal prior'
+
+
+    s = brainiak.reprsimil.brsa.GBRSA(SNR_prior='exp')
+    SNR_grids, SNR_weights = s._set_SNR_grids()
+    assert np.all(SNR_grids >= 0) and np.isclose(np.sum(SNR_weights), 1) and np.all(SNR_weights > 0) \
+        and np.all(np.diff(SNR_grids) > 0), \
+        'SNR_grids or SNR_weights not correct for exponential prior'
+
```

Comment:
Thanks! This is now added as `test_grid_flatten()` in test_gbrsa.py. Unfortunately this is a long test code, because we have to repeat many of the steps done by `_fit_RSA_marginalized` in the test code in order to set up all the tests. Additional check for flattening 2-D grids to 1-D grid is included in this function as well.